compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -4971,6 +4986,11 @@ private ParserRuleContext getNextRuleForVarName() {
                 return ParserRuleContext.ERROR_FIELD_MATCH_PATTERN_RHS;
             case RELATIVE_RESOURCE_PATH:
                 return ParserRuleContext.CLOSE_BRACKET;
+            case CHECKING_EXPRESSION:
+                parentCtx = getParentContext();
+                if (!(parentCtx == ParserRuleContext.QUERY_EXPRESSION)) {
```

Comment:
```suggestion
                if (parentCtx != ParserRuleContext.QUERY_EXPRESSION) {
```
btw, why do we need to check the query ctx here? Any sample?

Reply:
Removed this statement from the `BallerinaParserErrorHandler` and handled the failing case in the `BallerinaParser`